File: tests/test_config.f90
module test_config
    use kinds, only: dp
    use config, only: solver_config, load_config
    use test_support, only: assert_near, assert_true
    implicit none
contains
    subroutine run_config_tests()
        type(solver_config) :: cfg
        call load_config('tests/fixtures/sample_config.cfg', cfg)
        call assert_true(cfg%linear_size == 3, 'config linear_size')
        call assert_near(cfg%ode_t0, 0.0_dp, 1.0e-12_dp, 'config ode_t0')
        call assert_near(cfg%ode_t1, 4.0_dp, 1.0e-12_dp, 'config ode_t1')
        call assert_true(trim(cfg%ode_method) == 'euler', 'config ode_method')
    end subroutine run_config_tests
end module test_config

File: python_tests/test_config.py
import unittest

from python_solver.config import load_config


class ConfigTests(unittest.TestCase):
    def test_config_load(self) -> None:
        cfg = load_config("tests/fixtures/sample_config.cfg")
        self.assertEqual(cfg.linear_size, 3)
        self.assertAlmostEqual(cfg.ode_t0, 0.0)
        self.assertAlmostEqual(cfg.ode_t1, 4.0)
        self.assertEqual(cfg.ode_method, "euler")


if __name__ == "__main__":
    unittest.main()
